cypress/support/user.ts

```diff
@@ -39,19 +49,86 @@ const testCredentialStore = {
   },
 };
 
+function changeActiveRole(selectedRoleId: number) {
+  const token = window.localStorage.getItem('token');
+
+  if (!token) {
+    throw new Error('No logged in user');
+  }
+
+  const api = getE2EApi();
+  const request = api.selectRole({ selectedRoleId, token }).then((resp) => {
+    if (!resp.selectRole.token) {
+      return;
+    }
+
+    const { currentRole, user, exp } = jwtDecode(
+      resp.selectRole.token
+    ) as DecodedTokenData;
+
+    window.localStorage.setItem('token', resp.selectRole.token);
+    window.localStorage.setItem(
+      'currentRole',
+      currentRole.shortCode.toUpperCase()
+    );
+    window.localStorage.setItem('expToken', `${exp}`);
+    window.localStorage.setItem('user', JSON.stringify(user));
+  });
+
+  cy.wrap(request);
+}
+
+const externalTokenLogin = (
+  roleOrCredentials: 'user' | 'officer'
+): Cypress.Chainable<LoginMutation | ExternalTokenLoginMutation> => {
+  const credentials =
+    typeof roleOrCredentials === 'string'
+      ? testCredentialStoreStfc[roleOrCredentials]
+      : roleOrCredentials;
+
+  const api = getE2EApi();
+  const request = api.externalTokenLogin(credentials).then((resp) => {
+    if (!resp.externalTokenLogin.token) {
+      return resp;
+    }
+
+    const { user, exp } = jwtDecode(
+      resp.externalTokenLogin.token
+    ) as DecodedTokenData;
+
+    window.localStorage.setItem('token', resp.externalTokenLogin.token);
+    window.localStorage.setItem(
+      'currentRole',
+      roleOrCredentials === 'user' ? 'USER' : 'USER_OFFICER'
+    );
+    window.localStorage.setItem('expToken', `${exp}`);
+    window.localStorage.setItem('user', JSON.stringify(user));
+
+    return resp;
+  });
+
+  return cy.wrap(request);
+};
+
 const login = (
   roleOrCredentials:
     | 'user'
     | 'officer'
     | 'user2'
     | 'placeholderUser'
     | { email: string; password: string }
-): Cypress.Chainable<LoginMutation> => {
+): Cypress.Chainable<LoginMutation | ExternalTokenLoginMutation> => {
   const credentials =
     typeof roleOrCredentials === 'string'
       ? testCredentialStore[roleOrCredentials]
       : roleOrCredentials;
 
+  if (Cypress.env('STFC') === true) {
+    return externalTokenLogin(roleOrCredentials as 'user' | 'officer')
```

Comment:
I think this would be better handled in one of these 2 ways:
- We can add entries to `testCredentialStoreStfc` to map `user2` and `placeholderUser` to auth tokens, potentially reusing the same ones we already have. We can also check if `roleOrCredentials` is an object and pass `user` to `externalTokenLogin` if so. This would let us support all calls to the `login()` function in STFC mode, as long as the permissions match up
- If we don't want to support tests where `login()` is called with any values other than "user" or "officer", we should check `roleOrCredentials` before calling `externalTokenLogin()` and throw an exception if `roleOrCredentials` isn't "user" or "officer". This will warn us that we tried to run an unsupported test, rather than fail with a less clear reason later on

The main reason I'm asking for this is that the `roleOrCredentials as 'user' | 'officer'` conversion isn't necessarily safe, and it can lead to an error when looking up the token from `testCredentialStoreStfc` if `roleOrCredentials` wasn't one of those values .